src/AzSK.ADO/Framework/Core/SVT/ADO/ADO.Project.ps1

```diff
@@ -1004,4 +1004,281 @@ class Project: ADOSVTBase
+            $controlResult.LogException($_)
+        }
+
+         ## Checking AgentPools
+         $IsAgentPoolActive = $false
+         $thresholdLimit = $this.ControlSettings.AgentPool.AgentPoolHistoryPeriodInDays
+
+         # Fetch All Agent Pools
+         $url = "https://dev.azure.com/$($this.OrganizationContext.OrganizationName)/$($this.ResourceContext.ResourceName)/_apis/distributedtask/queues?api-version=6.0-preview.1"
+         try{
+             $res = [WebRequestHelper]::InvokeGetWebRequest($url);
+             $taskAgentQueues = @()
+             $flag = $false
+             if(($res | Measure-Object).Count -ne 0)
+             {
+                 # Filter out legacy agent pools (Hosted, Hosted VS 2017 etc.) as they are not visible to user on the portal.
+                $taskAgentQueues = $res | where-object{ ($_.pool.isLegacy -eq $false)};
+                if(($taskAgentQueues | Measure-Object).Count -ne 0)
+                {
+                    foreach ($AgentPool in $taskAgentQueues)
+                    {
+                        $url = "https://dev.azure.com/{0}/{1}/_settings/agentqueues?queueId={2}&__rt=fps&__ver=2" -f $($this.OrganizationContext.OrganizationName), $($this.ResourceContext.ResourceDetails.id) ,$AgentPool.id;
+                        $res = [WebRequestHelper]::InvokeGetWebRequest($url);
+                        if (([Helpers]::CheckMember($res[0], "fps.dataProviders.data") ) -and ($res[0].fps.dataProviders.data."ms.vss-build-web.agent-jobs-data-provider"))
+                        {
+                            #Filtering agent pool jobs specific to the current project.
+                            $agentPoolJobs = $res[0].fps.dataProviders.data."ms.vss-build-web.agent-jobs-data-provider".jobs | Where-Object {$_.scopeId -eq $this.ResourceContext.ResourceDetails.id};
+                            #Arranging in descending order of run time.
+                            $agentPoolJobs = $agentPoolJobs | Sort-Object queueTime -Descending
+                            #If agent pool has been queued at least once
+                            if (($agentPoolJobs | Measure-Object).Count -gt 0)
+                            {
+                                #Get the last queue timestamp of the agent pool
+                                if ([Helpers]::CheckMember($agentPoolJobs[0], "finishTime"))
+                                {
+                                    $agtPoolLastRunDate = $agentPoolJobs[0].finishTime;
+        
+                                    if ((((Get-Date) - $agtPoolLastRunDate).Days) -gt $thresholdLimit)
+                                    {
+                                        ## Inactive pool
+                                        $AgentPoolrow = New-Object psobject -Property $([ordered] @{"Resource Type"="AgentPool";"IsActive"="$($IsAgentPoolActive)"; "Additional Info" = "Agent pool has not been queued in the last $thresholdLimit days."})
```

Comment:
This object is getting created again at lin1 1195.
Please check

Reply:
removed duplicate creation of object